cro-clib/chain.h

```diff
@@ -66,6 +80,16 @@ CroResult cro_basic_restore_transfer_address(CroAddressPtr *address_out, const u
  */
 CroResult cro_basic_restore_viewkey(CroAddressPtr *address_out, const uint8_t *input);
 
+/**
+ * staked -> utxo
+ * tendermint_url: ws://localhost:26657/websocket
+ * user_data: tx data to send
+ * # Safety
+ */
+CroResult cro_broadcast(const char *tenermint_url_string,
+                        const uint8_t *user_data,
+                        uint32_t data_length);
```

Comment:
I have some worry about simply encapsulating everything(including network and storage) into the library, it will put many limitations on the user.
For example, on the mobile device, the user might need to use the native network stack, to handle platform-specific things, like permission, network status switch, etc.
Also, it can't be used in event-driven or async language or framework easily, like python's async, or nodejs.

Reply:
Yeah, there are two sides to this. I think we can stop with the C-bindings work at this stage, so that we don't overdo it, and wait for @ben-crypto feedback when they try to use it on iOS / Android -- either they'd want more of it (so that they don't need to worry about syncs, utxo management etc.) or less of it (due to the platform-specific things, as you mentioned)

Reply:
ok, i agree

Reply:
For those users who want the whole thing and don't care about networking stack and blocking IO,  we can encapsulate the `client-rpc` with a single API: `spawn_client_process`, they can access the functionality through JSON-RPC.

Reply:
the problem with JSON-RPC is that its HTTP port is open to other applications on the same host (or even the outside world if one isn't careful with settings and it's running on a machine with a public IP)

Reply:
> the problem with JSON-RPC is that its HTTP port is open to other applications on the same host (or even the outside world if one isn't careful with settings and it's running on a machine with a public IP)

We can support the UNIX domain socket then.

Reply:
Not sure if unix domain sockets can work on non-jailbroken iOS.

> In iOS 10, daemons/processes with the "seatbelt" entitlement (e.g. mediaserverd) don't allow you to create or connect to unix domain sockets anymore.

Reply:
> Not sure if unix domain sockets can work on non-jailbroken iOS.
> 
> > In iOS 10, daemons/processes with the "seatbelt" entitlement (e.g. mediaserverd) don't allow you to create or connect to unix domain sockets anymore.

Or we can expose the `client-rpc` as an single API like `fn client_execute(json_request) -> json_response`.